clang/test/CIR/CodeGen/same-name-struct.c

```diff
@@ -0,0 +1,34 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s
+
+typedef struct demo{
+    int memberOne;
+} demo;
+
+void testOne() {    
+    demo one = {1};
+}
+
+void testTwo() {
+    typedef struct demo {
+        int memberOne;
+        int memberTwo;
+    } demo;
+
+    demo two = {1, 2};
+}
+
+// CHECK: !ty_22demo22 = !cir.struct<struct "demo" {!cir.int<s, 32>}>
```

Comment:
Testcases should use regex for FileCheck, see example here: `clang/test/CIR/CodeGen/abstract-cond.c`